src/Ship.cpp

```diff
@@ -669,6 +669,12 @@ void Ship::SetFlightState(Ship::FlightState newState)
 		m_launchLockTimeout = std::min(std::max(2.0, 2.0 * (GetMass() / MASS_LOCK_REFERENCE)), 20.0);
 	}
 
+	if (newState == DOCKED) {
+		m_launchLockTimeout = 0.0;
```

Comment:
Oh: as now you could pass through docking surface, and it happens you "land" instead... Then if you ask for permission take-off your thruster remains active due to the "if" in ClearThrusterState.

So I would avoid this. But in general you needs not to dock again if your lock timeout is not expired...

So I leave it